File: java/src/main/java/com/crio/qcontest/repositories/ContestRepository.java
package com.crio.qcontest.repositories;

 import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.stream.Collectors;

import com.crio.qcontest.entities.Contest;
import com.crio.qcontest.entities.Level;

public class ContestRepository implements IContestRepository{
    private final Map<Long,Contest> contestMap;
    private Long autoIncrement = 1L;

    public ContestRepository(){
        contestMap = new HashMap<Long,Contest>();
    }

    @Override
    public Contest save(Contest contest) {
            Contest q = new Contest(contest.getTitle(),contest.getLevel(),contest.getCreator(),contest.getQuestions(),autoIncrement);
            contestMap.put(autoIncrement,q);
            ++autoIncrement;
            return q;
    }

    @Override
    public List<Contest> findAll() {
        return contestMap.values().stream().collect(Collectors.toList());
    }

    @Override
    public Optional<Contest> findById(Long id) {
        return Optional.ofNullable(contestMap.get(id));
    }

    // TODO: CRIO_TASK_MODULE_SERVICES
    // Complete the implementation of findAllContestLevelWise method
    // Implementation must take care of the following cases:-
    // 1) Find all the contests for a given level.

    @Override
    public List<Contest> findAllContestLevelWise(Level level) {
         return Collections.emptyList();
    }    
}

File: python/src/repositories/ContestRepository.py
from typing import List, Optional

from src.entities import Contest, Level

class ContestRepository:
    def __init__(self) -> None:
        self._contestMap = {}
        self._autoIncrement = 1

    def save(self,contest: Contest) -> Contest:
        c = Contest(contest.get_title(),contest.get_level(),contest.get_creator(),contest.get_questions(),self._autoIncrement)
        self._contestMap[self._autoIncrement] = c
        self._autoIncrement += 1
        return c

    def findAll(self) -> List[Contest]:
        return list(self._contestMap.values())

    def findById(self, id: int) -> Optional[Contest]:
        return self._contestMap.get(id)

    # TODO: CRIO_TASK_MODULE_SERVICES
    # Complete the implementation of findAllContestLevelWise method
    # Implementation must take care of the following cases:-
    # 1) Find all the contests for a given level.

    def findAllContestLevelWise(self,level: Level) -> List[Contest]:
        pass
